knowledge-base/grid-show-tooltip-on-column-header.md

```diff
@@ -0,0 +1,130 @@
+                </span>
+            </HeaderTemplate>
+        </GridColumn>
+
+        <GridColumn Field="@(nameof(SampleData.Name))" Width="165px">
+            <HeaderTemplate>
+                <span title="Long Employee Name"
+                      class="@(ShowTooltip.GetValueOrDefault(nameof(SampleData.Name), false) ? "employee-header" : "")">
+                    Long Employee Name
+                </span>
+            </HeaderTemplate>
+        </GridColumn>
+
+        <GridColumn Field="@(nameof(SampleData.Team))" Width="210px">
+            <HeaderTemplate>
+                <span title="Extremely Long Team Name"
+                      class="@(ShowTooltip.GetValueOrDefault(nameof(SampleData.Team), false) ? "employee-header" : "")">
+                    Extremely Long Team Name
+                </span>
+            </HeaderTemplate>
+        </GridColumn>
+    </GridColumns>
+</TelerikGrid>
+
+<TelerikTooltip TargetSelector=".employee-header" Class="my-callout">
+</TelerikTooltip>
+
+<style>
+    .my-callout .k-callout {
+        left: 30px !important;
+    }
+</style>
+
+@code {
+    private TelerikGrid<SampleData> GridRef { get; set; } = null!;
+
+    private IEnumerable<SampleData> MyData = Enumerable.Range(1, 10).Select(x => new SampleData
+        {
+            Id = x,
+            Name = "name " + x,
+            Team = "team" + x
+        });
+
+    // Define minimum width requirements for tooltips
+    private static readonly Dictionary<string, double> MinColumnWidths = new()
+    {
+        { nameof(SampleData.Id), 125 },
+        { nameof(SampleData.Name), 160 },
+        { nameof(SampleData.Team), 205 }
+    };
+
+    // Store whether a tooltip should be shown for each column
+    private Dictionary<string, bool> ShowTooltip = new();
+
+    private async Task HandleColumnWidthChange(GridStateEventArgs<SampleData> args)
+    {
+        foreach (var column in args.GridState.ColumnStates)
+        {
+            string columnField = column.Field;
+            if (MinColumnWidths.TryGetValue(columnField, out double minWidth))
+            {
+                double currentWidth = double.Parse(column.Width.Replace("px", ""));
+                ShowTooltip[columnField] = currentWidth < minWidth;
+            }
+        }
+
+        await GridRef.SetStateAsync(args.GridState);
+    }
+
+    public class SampleData
+    {
+        public int Id { get; set; }
+        public string Name { get; set; } = string.Empty;
+        public string Team { get; set; } = string.Empty;
+    }
+}
+````
+
+> The additional CSS is used to adjust the position of the tooltip callout. Modify this CSS based on your application's specific layout and design requirements.
+
+## See Also
+- [Grid Column Header Template Documentation](slug:components/grid/templates/column-header#column-header-template)
+- [Telerik Tooltip Overview](slug:components/tooltip/overview)
+- [Tooltip Template Feature](slug:components/tooltip/template)
```

Comment:
Tooltip templates are not used in the KB, so we can remove this link for brevity.